src/components/Clock/Clock.tsx

```diff
@@ -0,0 +1,59 @@
+import { Component } from 'react';
+
+import './Clock.scss';
+
+type ClockName = number | string;
+
+interface Props {
+  name: ClockName;
+}
+
+interface State {
+  clockName: ClockName;
+  date: Date;
+}
+
+export class Clock extends Component<Props, State> {
+  state: State = {
+    clockName: this.props.name,
+    date: new Date(),
+  };
+
+  timerId: NodeJS.Timer = setTimeout(() => {}, 0);
+
+  componentDidMount() {
+    this.timerId = setInterval(() => {
+      this.setState({ date: new Date() });
+      // eslint-disable-next-line no-console
+      console.log(this.state.date.toLocaleTimeString());
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    this.state.clockName = this.props.name;
+
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line no-console
+      console.log(`The Clock was renamed from ${prevProps.name}`
+        + ` to ${prevState.clockName}`);
+    }
+  }
```

Comment:
You don't even need state here:
```suggestion
  componentDidUpdate(prevProps: Props) {
    if (prevProps.name !== this.props.name) {
      // eslint-disable-next-line no-console
      console.log(`The Clock was renamed from ${prevProps.name}`
        + ` to ${this.props.name}`);
    }
  }
```